tests/unit/utils_math.cpp

```diff
@@ -21,3 +21,127 @@ TEST(UtilsMath, Log2) {
     ASSERT_EQ(memgraph::utils::Log2(i), static_cast<uint64_t>(log2(i)));
   }
 }
+
+TEST(UtilsMath, IdenticalPositiveValues) {
+  const auto epsilon = memgraph::utils::GetEpsilon();
+  const auto a = 1.0;
+  const auto b = 1.0;
+
+  EXPECT_TRUE(memgraph::utils::CompareDouble(a, b, epsilon) == 0);
+}
+
+TEST(UtilsMath, IdenticalNegativeValues) {
+  const auto epsilon = memgraph::utils::GetEpsilon();
+  const auto a = -1.0;
+  const auto b = -1.0;
+
+  EXPECT_TRUE(memgraph::utils::CompareDouble(a, b, epsilon) == 0);
+}
+
+TEST(UtilsMath, EpsilonTooSmallSoExpectingDeath) {
+  const auto epsilon = memgraph::utils::GetEpsilon() / 10;
+
+  EXPECT_DEATH(memgraph::utils::CompareDouble(0.0, 0.0, epsilon), "");
+}
+
+TEST(UtilsMath, FirstValueMuchSmallerThanSecond) {
+  const auto a = 1.0;
+  const auto b = 2.0;
+
+  EXPECT_TRUE(memgraph::utils::CompareDouble(a, b) < 0);
+}
+
+TEST(UtilsMath, FirstValueMuchGreaterThanSecond) {
+  const auto a = 2.0;
+  const auto b = 1.0;
+
+  EXPECT_TRUE(memgraph::utils::CompareDouble(a, b) > 0);
+}
+
+TEST(UtilsMath, FirstValueSmallerThanButEqualToSecond) {
+  const auto epsilon = memgraph::utils::GetEpsilon();
+  const auto a = 1.0;
+  const auto b = a + epsilon / 2;
+
+  EXPECT_TRUE(memgraph::utils::CompareDouble(a, b) == 0);
+}
+
+TEST(UtilsMath, FirstValueGreaterThanButEqualToSecond) {
+  const auto epsilon = memgraph::utils::GetEpsilon();
+  const auto a = 2.0;
+  const auto b = a - epsilon / 2;
+
+  EXPECT_TRUE(memgraph::utils::CompareDouble(a, b) == 0);
+}
+
+TEST(UtilsMath, FirstValueSmallerThanAndNotEqualToSecond) {
+  const auto epsilon = memgraph::utils::GetEpsilon();
+  const auto a = 1.0;
+  const auto b = a + 2 * epsilon;
+
+  EXPECT_TRUE(memgraph::utils::CompareDouble(a, b) < 0);
+}
+
+TEST(UtilsMath, FirstValueGreaterThanAndNotEqualToSecond) {
+  const auto epsilon = memgraph::utils::GetEpsilon();
+  const auto a = 2.0;
+  const auto b = a - 2 * epsilon;
+
+  EXPECT_TRUE(memgraph::utils::CompareDouble(a, b) > 0);
+}
+
+TEST(UtilsMath, IsStrictlyGreaterBigDistance) {
+  const auto a = 2.0;
+  const auto b = 1.0;
+
+  EXPECT_TRUE(memgraph::utils::IsStrictlyGreater(a, b));
+}
+
+TEST(UtilsMath, IsStrictlyGreaterWithSmallDistance) {
+  const auto a = 1.0;
+  const auto b = 1.0 - 2 * memgraph::utils::GetEpsilon();
+
+  EXPECT_TRUE(memgraph::utils::IsStrictlyGreater(a, b));
+}
+
+TEST(UtilsMath, IsStrictlyGreaterTooSmallDistance) {
+  const auto a = 1.0;
+  const auto b = 1.0 - memgraph::utils::GetEpsilon() / 2;
+
+  EXPECT_TRUE(!memgraph::utils::IsStrictlyGreater(a, b));
```

Comment:
```suggestion
  EXPECT_FALSE(memgraph::utils::IsStrictlyGreater(a, b));
```

It is easy to not notice the exclamation mark, so being verbose here and using `EXPECT_FALSE(...)` instead of `EXPECT_TRUE(!...)` can be beneficial.

Even though in this case it doesn't really matter because of the simplicity of the tests, but if you have to modify something else on the PR, it might also worth to update these checks.